### PR TITLE
[kmac] Entropy Shuffle

### PR DESCRIPTION
**Pull request by eunchan**
After prepping the 320-bit entropy, the KMAC now shuffles the
concatenated entropies.

Previously, 320b entropy is concatenated five times to make 1600b
entropy that is used in the Keccak DOM logic. By shuffling, the 1600b
does not have duplicated pattern using the compile-time random constant
parameter.

### CODE REVIEW

File: hw/ip/kmac/data/kmac.hjson [hunk 1]
```diff
@@ -126,6 +126,12 @@
       randcount: "64"
       randtype:  "perm"
     }
+    { name:      "RndCnstStoragePerm"
+      desc:      "Compile-time random permutation for Storage Entropy"
+      type:      "kmac_pkg::storage_perm_t"
+      randcount: "1600"
+      randtype:  "perm"
```
Comment (tjaychen) [inline hw/ip/kmac/data/kmac.hjson:133]: this will get exposed to the top level right? this is probably something we want nuvoton to regenerate.

**Review by tjaychen — APPROVED** [review]

File: hw/ip/kmac/data/kmac.hjson (hunk 1 above)
Comment (eunchan) [reply hw/ip/kmac/data/kmac.hjson:133]: Ah.. yep. I should re-generate top here too.
A few more random constants for KMAC has been added. I will send an email to them

Comment (eunchan) [reply hw/ip/kmac/data/kmac.hjson:133]: CC: @arnonsha

**Comment by eunchan** [comment]
Merging now...
image

**Comment by arnonsha** [comment]
> LGTM
> 
> @arnonsha can correct me if I am wrong, but I believe Nuvoton already has a sort of "release" script in place that triggers regeneration of all rand constants (including LC and OTP).

Yes, we do have one

**Pull request merged**